### DISCUSSION TITLE
performance of  carla 0.9.12

### DISCUSSION THREAD

**Discussion by ynlch** [carla-simulator/carla#4956] [Q&A]
I attach 8 rbg camera to one car, but I find each camera's frame rate is only 8fps, but GPU only 45%, CPU about 50%, PCIe bandwidth only 6%. 
I'm running  carla 0.9.12 with nvida RTX 3070, intel I7-11700K@3.6GHz x 16, 32G DDR. 
with simple debug, I find function InRHICmdList.ReadSurfaceData() about 20ms each frame, 
does anyone know the reason and how to fix it? I want to keep GUP and CPU full used

**Comment by gengtsh** [comment]
can you please tell me how you debug and trace the performance ? thanks
